packages/editor-ui/src/components/InlineExpressionEditor/__tests__/InlineExpressionTip.test.ts

```diff
@@ -0,0 +1,121 @@
+import { createTestingPinia } from '@pinia/testing';
+import { renderComponent } from '@/__tests__/render';
+import InlineExpressionTip from '@/components/InlineExpressionEditor/InlineExpressionTip.vue';
+import type { useNDVStore } from '@/stores/ndv.store';
+import { EditorSelection, EditorState } from '@codemirror/state';
+import type { CompletionResult } from '@codemirror/autocomplete';
+import { FIELDS_SECTION } from '@/plugins/codemirror/completions/constants';
+
+let mockNdvState: Partial<ReturnType<typeof useNDVStore>>;
+let mockCompletionResult: Partial<CompletionResult>;
+
+vi.mock('@/stores/ndv.store', () => {
+	return {
+		useNDVStore: vi.fn(() => mockNdvState),
+	};
+});
+
+vi.mock('@/plugins/codemirror/completions/datatype.completions', () => {
+	return {
+		datatypeCompletions: vi.fn(() => mockCompletionResult),
+	};
+});
+
+describe('InlineExpressionTip.vue', () => {
+	beforeEach(() => {
+		mockNdvState = {
+			hasInputData: true,
+			isDNVDataEmpty: vi.fn(() => true),
+		};
+	});
+
+	test('should show the default tip', async () => {
+		const { container } = renderComponent(InlineExpressionTip, {
+			pinia: createTestingPinia(),
+		});
+		expect(container).toHaveTextContent('Tip: Anything inside {{ }} is JavaScript. Learn more');
+	});
+
+	describe('When the NDV input is not empty and a mappable input is focused', () => {
+		test('should show the drag-n-drop tip', async () => {
+			mockNdvState = {
+				hasInputData: true,
+				isDNVDataEmpty: vi.fn(() => false),
+				focusedMappableInput: 'Some Input',
+			};
+			const { container } = renderComponent(InlineExpressionTip, {
+				pinia: createTestingPinia(),
+			});
+			expect(container).toHaveTextContent('Tip: Drag aninput fieldfrom the left to use it here.');
```

Comment:
The expectation will fail with whitespaces because they come from styling?

Reply:
yes, they're in separate html elements so .textContent sticks them together (but in reality CSS adds spacing)